gnovm/pkg/gnolang/type_check.go

```diff
@@ -215,6 +215,122 @@ func assertAssignableTo(xt, dt Type, autoNative bool) {
 	}
 }
 
+func checkConstantExpr(store Store, last BlockNode, vx Expr) {
+Main:
+	switch vx := vx.(type) {
+	case *NameExpr:
+		t := evalStaticTypeOf(store, last, vx)
+		if _, ok := t.(*ArrayType); ok {
```

Comment:
doubt with this, only with ten(arr)?

Reply:
consider this:
```go
package main

func main() {
	const a = [2]int{1, 2}
	println(a)
}
```
and 
```go
package main

func main() {
	a := [2]int{1, 2}

	const b = a
}
```
these should not work, but works on the current branch.

Reply:
nice catch I did a bad merge ^^

Reply:
hmmm, this logic seems incorrect:
```go
if _, ok := t.(*ArrayType); ok {
	break Main
}
```

Due to this logic, the error in `const43.gno` was not caught.

I guess your intention is to check len(arr), right? I think that needs additional context to know if the last expr is a built func call: `len` or `max`?

Not so sure, but we need to focus on ensuring that the logic of this function is correct.